### PR TITLE
Add array example code with Go

### PR DESCRIPTION
**Pull request by nosqlcoco**
in my Go code. using the build-in functions of go, replace for loop, like copy and append keywords.
if not easy to understant. I can repalce it with for loop.

If this PR is related to coding or code translation, please fill out the checklist.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/go/chapter_array_and_linkedlist/my_array.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+// File: my_array.go
```
Comment (krahets) [inline codes/go/chapter_array_and_linkedlist/my_array.go:1]: The file name should be `array.go`. Do not change the file name.

File: codes/go/chapter_array_and_linkedlist/my_array.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	return res
+}
+
+/* 在数组的索引 index 处插入元素 num */
+func insert(nums []int, num int, index int) []int {
+	// 初始化数组
+	var res = []int{}
+	// 截取nums index前的元素放入res
+	// append是Go的内置函数,用于往数组中追加元素
+	res = append(res, nums[:index]...)
```
Comment (krahets) [inline codes/go/chapter_array_and_linkedlist/my_array.go:40]: Please use for loop.

File: codes/go/chapter_array_and_linkedlist/my_array_test.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// File: my_array_test.go
```
Comment (krahets) [inline codes/go/chapter_array_and_linkedlist/my_array_test.go:1]: array_test.go

File: codes/go/chapter_array_and_linkedlist/my_array.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+func traverse(nums []int) {
+	var count = 0
+	// 通过索引遍历数组
+	for i := 0; i < len(nums); i++ {
+		count++
+	}
+	// 直接遍历数组
+	for i, v := range nums {
+		count++
+		// 这里辅助打印变量i和v, 因为在Go中如果申明变量而不使用无法编译通过
```
Comment (krahets) [inline codes/go/chapter_array_and_linkedlist/my_array.go:70]: 声明

File: codes/go/chapter_array_and_linkedlist/my_array.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	// 截取nums index后的元素放入res
+	res = append(res, nums[index:]...)
+	return res
+}
+
+/* 删除索引 index 处元素 */
+func remove(nums []int, index int) []int {
+	// 初始化数组
+	var res = []int{}
+	// 截取nums index前的元素放入res
```
Comment (krahets) [inline codes/go/chapter_array_and_linkedlist/my_array.go:52]: Please use the comment copying from the reference language.

**Review by krahets — COMMENTED** [review]
Thank @nosqlcoco for the code. Please address the comments.

It is recommended to make the code and comment similar to the reference language as far as possible. We want to keep the coherence of multiple languages.

**Comment by guowei-gong** [comment]
I think this is an example of slice, not arrays

**Comment by krahets** [comment]
This PR is closed because of no response from the author.

**Pull request closed without merging**